tests/test_micrograph.py

```diff
@@ -0,0 +1,307 @@
+    """
+    Test the MicrographSource has the correct values from arguments.
+    """
+    s = sim_fixture
+    m = micrograph_fixture
+    assert s.L == m.particle_box_size
+    assert s == m.simulation
+    assert m.particles_per_micrograph * m.micrograph_count <= s.n
+    assert len(m) == m.micrograph_count
+    assert m.clean_micrographs[0].shape[1] == m.micrograph_size
+    assert m.clean_micrographs[0].shape[2] == m.micrograph_size
+    assert (
+        repr(m)
+        == f"{m.micrograph_count} {m.dtype.name} micrographs of size {m.micrograph_size}x{m.micrograph_size}"
+    )
+    assert np.array_equal(s.images[0], m.images[0])
+    _ = m.clean_micrographs[:]
+    _ = m.micrographs[:]
+
+
+def test_micrograph_raises_error_simulation():
+    """
+    Test that MicrographSource raises error when simulation argument is not a Simulation
+    """
+    with pytest.raises(Exception) as e_info:
+        _ = MicrographSource(
+            "Simulation",
+            micrograph_size=100,
+            particles_per_micrograph=20,
+            interparticle_distance=10,
+        )
+    assert str(e_info.value) == "Simulation should be of type Simulation."
+
+
+def test_micrograph_raises_error_image_size(sim_fixture):
+    """
+    Test the MicrographSource class raises errors when the image size is larger than micrograph size.
+    """
+    with pytest.raises(ValueError) as e_info:
+        s = sim_fixture
+        _ = MicrographSource(
+            s,
+            micrograph_size=s.L - 1,
+            particles_per_micrograph=10,
+            interparticle_distance=0,
+        )
+    assert (
+        str(e_info.value)
+        == "The micrograph size must be larger or equal to the simulation's image size."
+    )
+
+
+def test_micrograph_raises_error_image_count(sim_fixture):
+    """
+    Test the MicrographSource class raises errors when there aren't enough images.
+    """
+    with pytest.raises(ValueError) as e_info:
+        s = sim_fixture
+        _ = MicrographSource(
+            s,
+            micrograph_size=s.L * 10,
+            particles_per_micrograph=s.n + 1,
+        )
+    assert (
+        str(e_info.value)
+        == "The particle count times micrograph count must be less than or equal to the number o\
+f images generated by the simulation (particles_per_micrograph * micrograph_count <= simulation.n)"
+    )
+
+
+def test_micrograph_centers_match(micrograph_fixture):
+    """
+    Test that the Micrograph's centers are forming at generated points.
+    """
+    m = micrograph_fixture
+    centers = np.reshape(m.centers, (m.total_particle_count, 2))
+    for i, center in enumerate(centers):
+        if (
+            center[0] >= 0
+            and center[0] < m.micrograph_size
+            and center[1] >= 0
+            and center[1] < m.micrograph_size
+        ):
+            assert m.clean_micrographs[i // m.particles_per_micrograph].asnumpy()[0][
```

Comment:
I don't follow. Why should this not be equal to the max?

Reply:
Since we're subtracting the values of the image, the actual value of the images on the micrograph will be lower than the surroundings (which should be the max without noise). The test doesn't sound too effective but works okay in practice